src/history/HistoryArchiveManager.cpp

```diff
@@ -176,10 +178,23 @@ HistoryArchiveManager::scheduleHistoryArchiveReportWork() const
             mApp, 0, archive, "archive-report-" + archive->getName(),
             BasicWork::RETRY_NEVER));
     }
-    return mApp.getWorkScheduler().scheduleWork<HistoryArchiveReportWork>(
-        hasWorks);
+    return std::make_shared<HistoryArchiveReportWork>(mApp, hasWorks);
 };
 
+std::shared_ptr<BasicWork>
+HistoryArchiveManager::getCheckLedgerHeaderWork(
+    LedgerHeaderHistoryEntry const& lhhe) const
+{
+    std::vector<std::shared_ptr<BasicWork>> checkWorks;
+    for (auto const& archive : mArchives)
+    {
+        checkWorks.emplace_back(
+            std::make_shared<CheckSingleLedgerHeaderWork>(mApp, archive, lhhe));
```

Comment:
note that `WorkSequence` stops on first failure

Reply:
Yeah, I think that is actually what we want from a diagnostic command.